docker_daemon/test_docker_daemon.py

```diff
@@ -652,3 +652,33 @@ def test_container_name_extraction(self):
         ]
         for co in containers:
             self.assertEqual(DockerUtil.container_name_extractor(co[0]), co[1])
+
+    def test_collect_exit_code(self):
+        config = {
+            "init_config": {},
+            "instances": [{
+                "url": "unix://var/run/docker.sock",
+                "collect_exit_codes": True
+            }]
+        }
+        DockerUtil().set_docker_settings(config['init_config'], config['instances'][0])
+
+        expected_metrics = [
+            ("docker.containers.exit", ['docker_image:nginx:latest', 'image_name:nginx', 'image_tag:latest', 'container_name:test-exit-code', 'exit_code:1'])
+        ]
+
+        container = self.docker_client.create_container(
+            "nginx:latest", detach=True, name='test-exit-code', entrypoint='/bin/false')
+        log.debug('start nginx:latest with entrypoint /bin/false')
+        self.docker_client.start(container)
+        log.debug('container exited with %s' % self.docker_client.wait(container, 1))
+        # After the container exits, we need to wait a second so the event isn't too recent
+        # when the check runs, otherwise the event is not picked up
+        from time import sleep
+        sleep(1)
+
+        self.run_check(config)
```

Comment:
When the test runs `run_check`, the DIE event of the container just appeared. The check will collect the events that appeared between 60 seconds ago and the second it started running, *excluded*. Since the DIE event likely appeared in the same second the check ran, it will not be picked up and so the check will fail. The `sleep(1)` makes sure the event isn't too recent when the check runs. Another option, which I don't think is worth the trouble, would be to make a small change in `dockerutils.py` so even the most recent events are picked up.
What do you think @hkaj ?